Back-To-School/Assets/BackToSchool/Scripts/GameManagement/Game.cs

```diff
@@ -0,0 +1,179 @@
+            _pausePresenter.MenuReturned += ReturnToMenu;
+
+            _player.WeaponController.AmmoChanged    += _hudPresenter.OnAmmoChanged;
+            _player.WeaponController.WeaponChanged  += _hudPresenter.OnWeaponChanged;
+            _player.WeaponController.MaxAmmoChanged += _hudPresenter.OnMaxAmmoChanged;
+            _player.Died                            += OnPlayerDeath;
+            _player.HealthChanged                   += _hudPresenter.OnHealthChanged;
+            _enemySpawner.EnemyDied                 += _levelSystem.AddExperience;
+
+            _levelSystem.LevelChanged      += _statsManager.LevelUp;
+            _levelSystem.LevelChanged      += _hudPresenter.OnLevelChanged;
+            _levelSystem.ExperienceChanged += _hudPresenter.OnExpChanged;
+            _saveAction                    =  delegate { _saveSystem.SavePlayerProgress(_player, _levelSystem); };
+            _levelSystem.ProgressChanged   += _saveAction;
+
+            _statsManager.ArmorChanged     += _hudPresenter.OnArmorChanged;
+            _statsManager.DamageChanged    += _hudPresenter.OnDamageChanged;
+            _statsManager.MaxHealthChanged += _hudPresenter.OnMaxHealthChanged;
+            _statsManager.MoveSpeedChanged += _hudPresenter.OnMoveSpeedChanged;
+        }
+
+        private void OnDestroy()
+        {
+            _gameOverPresenter.Restarted -= RestartGame;
+            _pausePresenter.Restarted    -= RestartGame;
+            _pausePresenter.Continued    -= ContinueGame;
+            _pausePresenter.MenuReturned -= ReturnToMenu;
+
+            _player.WeaponController.AmmoChanged    -= _hudPresenter.OnAmmoChanged;
+            _player.WeaponController.WeaponChanged  -= _hudPresenter.OnWeaponChanged;
+            _player.WeaponController.MaxAmmoChanged -= _hudPresenter.OnMaxAmmoChanged;
+            _player.Died                            -= OnPlayerDeath;
+            _player.HealthChanged                   -= _hudPresenter.OnHealthChanged;
+            _enemySpawner.EnemyDied                 -= _levelSystem.AddExperience;
+
+            _levelSystem.LevelChanged      -= _statsManager.LevelUp;
+            _levelSystem.LevelChanged      -= _hudPresenter.OnLevelChanged;
+            _levelSystem.ExperienceChanged -= _hudPresenter.OnExpChanged;
+            _levelSystem.ProgressChanged   -= _saveAction;
+
+            _statsManager.ArmorChanged     -= _hudPresenter.OnArmorChanged;
+            _statsManager.DamageChanged    -= _hudPresenter.OnDamageChanged;
+            _statsManager.MaxHealthChanged -= _hudPresenter.OnMaxHealthChanged;
+            _statsManager.MoveSpeedChanged -= _hudPresenter.OnMoveSpeedChanged;
+        }
+
+        #region GameHandlers
+
+        private void OnPlayerDeath()
+        {
+            _isPlayerDead = true;
+            _enemySpawner.SetTarget(null);
+            Invoke(nameof(EndGame), _gameOverDelay);
+        }
+
+        private void StopGame()
```

Comment:
reorder please these 3 methods: OnGameStopped, Continue, Stop.
Also, OnGameStopped may be renamed to OnGamePaused.